### PR TITLE
Create bug-fix release 0.26.1

### PR DESCRIPTION
**Pull request by AmintorDusko**
**Context:** There is an issue with the returned probability in some edge cases.

**Description of the Change:** 
* The transposition step will be fixed.
* Probability integration tests expanded.
* Transposition unit tests included.

**Benefits:** Right results.

**Possible Drawbacks:**

**Related GitHub Issues:**
#375

### CODE REVIEW

**Comment by AmintorDusko** [comment]
[sc-27673]

**Comment by multiphaseCFD** [comment]
Hi @AmintorDusko , nice work! Just want to let you know that `std::ranges::reverse_view` doesn't work on Mac M1 that I'm using now.

**Comment by AmintorDusko** [comment]
> Hi @AmintorDusko , nice work! When compiling you branch, I got following errors: `error: no member named 'reverse_view' in namespace 'std::ranges'`. Maybe rolling back to your previous implementation will fix the issue.

Hi @multiphaseCFD, the work is not done yet. This is a draft pull request.

**Comment by multiphaseCFD** [comment]
Thanks @AmintorDusko for putting these together! Just one mirror comments.

File: pennylane_lightning/src/tests/Test_Util.cpp [hunk 1]
```diff
@@ -276,3 +276,49 @@ TEST_CASE("Test AlignedAllocator", "[Util][Memory]") {
...
+        {{3, 0, 1, 2}, {0, 2, 4, 6, 8, 10, 12, 14, 1, 3, 5, 7, 9, 11, 13, 15}},
+        {{3, 0, 2, 1}, {0, 4, 2, 6, 8, 12, 10, 14, 1, 5, 3, 7, 9, 13, 11, 15}},
+        {{3, 1, 0, 2}, {0, 2, 8, 10, 4, 6, 12, 14, 1, 3, 9, 11, 5, 7, 13, 15}},
+        {{3, 1, 2, 0}, {0, 8, 2, 10, 4, 12, 6, 14, 1, 9, 3, 11, 5, 13, 7, 15}},
+        {{3, 2, 0, 1}, {0, 4, 8, 12, 2, 6, 10, 14, 1, 5, 9, 13, 3, 7, 11, 15}},
+        {{3, 2, 1, 0}, {0, 8, 4, 12, 2, 10, 6, 14, 1, 9, 5, 13, 3, 11, 7, 15}}};
+
+    SECTION("Looping over different wire configurations:") {
+        for (const auto &term : input) {
+            // Defining a vector to be transposed.
```
Comment (multiphaseCFD) [inline pennylane_lightning/src/tests/Test_Util.cpp:316]: what about tensor here?

Comment (AmintorDusko) [reply pennylane_lightning/src/tests/Test_Util.cpp:316]: I didn't get it. Could you please expand on your question?

**Comment by multiphaseCFD** [comment]
> I didn't get it. Could you please expand on your question? 

What about "Defining a vector to be transposed." -> "Defining a tensor to be transposed." ?

File: pennylane_lightning/src/tests/Test_Util.cpp [hunk 2]
```diff
@@ -276,3 +276,49 @@ TEST_CASE("Test AlignedAllocator", "[Util][Memory]") {
...
+        {{2, 1, 0, 3}, {0, 1, 8, 9, 4, 5, 12, 13, 2, 3, 10, 11, 6, 7, 14, 15}},
+        {{2, 1, 3, 0}, {0, 8, 1, 9, 4, 12, 5, 13, 2, 10, 3, 11, 6, 14, 7, 15}},
+        {{2, 3, 0, 1}, {0, 4, 8, 12, 1, 5, 9, 13, 2, 6, 10, 14, 3, 7, 11, 15}},
+        {{2, 3, 1, 0}, {0, 8, 4, 12, 1, 9, 5, 13, 2, 10, 6, 14, 3, 11, 7, 15}},
+        {{3, 0, 1, 2}, {0, 2, 4, 6, 8, 10, 12, 14, 1, 3, 5, 7, 9, 11, 13, 15}},
+        {{3, 0, 2, 1}, {0, 4, 2, 6, 8, 12, 10, 14, 1, 5, 3, 7, 9, 13, 11, 15}},
+        {{3, 1, 0, 2}, {0, 2, 8, 10, 4, 6, 12, 14, 1, 3, 9, 11, 5, 7, 13, 15}},
+        {{3, 1, 2, 0}, {0, 8, 2, 10, 4, 12, 6, 14, 1, 9, 3, 11, 5, 13, 7, 15}},
+        {{3, 2, 0, 1}, {0, 4, 8, 12, 2, 6, 10, 14, 1, 5, 9, 13, 3, 7, 11, 15}},
+        {{3, 2, 1, 0}, {0, 8, 4, 12, 2, 10, 6, 14, 1, 9, 5, 13, 3, 11, 7, 15}}};
```
Comment (mlxd) [inline pennylane_lightning/src/tests/Test_Util.cpp:312]: This must've been fun to write.

File: pennylane_lightning/src/util/Util.hpp [hunk 3]
```diff
@@ -312,8 +312,10 @@ inline auto transposed_state_index(size_t ind,
                                    const std::vector<size_t> &new_axes)
     -> size_t {
     size_t new_index = 0;
-    for (size_t axis : new_axes) {
-        new_index += (ind % 2) << axis;
+    const size_t max_axis = new_axes.size() - 1;
+    // NOLINTNEXTLINE(modernize-loop-convert)
```
Comment (mlxd) [inline pennylane_lightning/src/util/Util.hpp:316]: Why do we not want this modernized?

File: pennylane_lightning/src/util/Util.hpp [hunk 4]
```diff
@@ -334,7 +336,7 @@ auto transpose_state_tensor(const std::vector<T> &tensor,
     -> std::vector<T> {
     std::vector<T> transposed_tensor(tensor.size());
     for (size_t ind = 0; ind < tensor.size(); ind++) {
-        transposed_tensor[transposed_state_index(ind, new_axes)] = tensor[ind];
+        transposed_tensor[ind] = tensor[transposed_state_index(ind, new_axes)];
```
Comment (mlxd) [inline pennylane_lightning/src/util/Util.hpp:339]: So the order of assignment was wrong previously?

**Review by mlxd — APPROVED** [review]
Thanks Amintor. Just some questions, but not blocking the approval.

File: pennylane_lightning/src/util/Util.hpp (hunk 3 above)
Comment (AmintorDusko) [reply pennylane_lightning/src/util/Util.hpp:316]: Because the proposed change uses `std::ranges::reverse_view`, and this is not working fine on MacOS.
Also Tidy will not be happy with his own suggestion, saying that:
`no viable constructor or deduction guide for deduction of template arguments of 'reverse_view' [clang-diagnostic-error]`
You can see more here:
https://github.com/PennyLaneAI/pennylane-lightning/actions/runs/3245244628/jobs/5322519450

File: pennylane_lightning/src/util/Util.hpp (hunk 4 above)
Comment (AmintorDusko) [reply pennylane_lightning/src/util/Util.hpp:339]: I remade from scratch, this part and refactor two parts of the code, this is one of them. The other one is the method `transposed_state_index`.
I didn't go too deep into why it was not working before, I have just a vague idea, but you can define it in two ways.
Like it was, and like it is now. But the indexes need to be right, and they were not always right before.

File: pennylane_lightning/src/util/Util.hpp (hunk 3 above)
Comment (AmintorDusko) [reply pennylane_lightning/src/util/Util.hpp:316]: Yes. I think it makes sense.

**Review by multiphaseCFD — APPROVED** [review]
